src/main/java/com/minecolonies/coremod/blocks/BlockTimberFrame.java

```diff
@@ -63,104 +52,109 @@ private void initBlock(final String name)
         setHardness(BLOCK_HARDNESS);
         setResistance(RESISTANCE);
     }
-
+    
+    /**
+     * Calc the default state depending on the neighboring blocks.
+     * @deprecated remove this when not needed anymore
+     * @param state
+     * @param world
+     * @param pos
+     * @return
+     */
+    @Deprecated
     @Override
     public IBlockState getActualState(final IBlockState state, final IBlockAccess world, final BlockPos pos)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "getActualState" is 24 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)